lib/puppet/agent.rb

```diff
@@ -43,9 +43,11 @@ def run(client_options = {})
     end
 
     result = nil
+    wait_for_lock_deadline = nil
```

Comment:
Is this line needed?

Reply:
It worked without this line, but as Josh suggested, I think the variable might reset itself this each time a LockError is raised. So it's just some extra protection.